src/main/java/games/strategy/engine/vault/VaultID.java

```diff
@@ -46,7 +47,7 @@ public boolean equals(final Object o) {
 
   @Override
   public int hashCode() {
-    return ((int) m_uniqueID) ^ m_generatedOn.getName().hashCode();
+    return Objects.hash(m_uniqueID, m_generatedOn.getName());
```

Comment:
I'm okay leaving this as-is because it is existing behavior.

However, I wanted to point out that the `VaultID#equals()` implementation uses `m_generatedOn` in the equality check, not `m_generatedOn.getName()`.  In fact, `Node` (the only `INode` implementation) doesn't even use the node name in its `equals()` or `hashCode()` implementations; it only uses the node's address and port.  So the above hash code definition pretty much breaks the `equals()`/`hashCode()` contract for `VaultID`, but it's probably been in the code forever, so there may be some unanticipated consequences to change it now.